aten/src/ATen/native/mkldnn/xpu/Blas.cpp

```diff
@@ -512,4 +512,54 @@ TORCH_IMPL_FUNC(addmv_out_xpu)
   xpu::addmv_out(self, mat, vec, beta, alpha, const_cast<Tensor&>(result));
 }
 
-} // namespace at::native
\ No newline at end of file
+Tensor _weight_int4pack_mm_xpu(
+    const Tensor& A,
+    const Tensor& B,
+    int64_t qGroupSize,
+    const Tensor& qScale,
+    const Tensor& qZeros) {
+  constexpr int64_t kNTileSize = 8;
+
+  auto M = A.size(0); // M
+  auto N = B.size(0) * kNTileSize; // N1=LCM(N, K)
+  auto K = A.size(1); // K1=LCM(K, 1024)
+  TORCH_CHECK(
+      A.dtype() == kBFloat16 || A.dtype() == kHalf || A.dtype() == kFloat,
+      __func__,
+      " : expect A to be either 32-bit or 16-bit float tensor.");
+  TORCH_CHECK(A.is_contiguous(), __func__, " : expect A to be contiguous.");
+  TORCH_CHECK(A.dim() == 2, __func__, " : expect A to be 2D tensor.");
+
+  TORCH_CHECK(B.dtype() == kInt, __func__, " : expect B to be int32 tensor.");
```

Comment:
In OneDNN implementation, you shall check zero points dtype as well. see https://oneapi-src.github.io/oneDNN/dev_guide_op_dynamicdequantize.html?highlight=s4#supported-data-types